app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,33 @@
+class QueueItemsController < ApplicationController
+  before_action :require_user
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+    queue_video(video)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.destroy if current_user.queue_items.include?(queue_item)
+    redirect_to my_queue_path
+  end
+
+  private
+  
+  def queue_video(video)
+    QueueItem.create(video: video, user: current_user, position: new_queue_item_position) unless current_user_queue_video?(video)
```

Comment:
I would recommend changing this to a three line conditional statement, since the line is so long. Try to keep your lines to about 80 characters.